agent/handlers/task_server_setup_sudo_integ_test.go

```diff
@@ -0,0 +1,427 @@
+		state.EXPECT().ContainerMapByArn(taskARN).Return(containerNameToDockerContainer, true).AnyTimes()
+		state.EXPECT().TaskByArn(taskARN).Return(task, true).AnyTimes()
+		state.EXPECT().ContainerByID(containerID).Return(dockerContainer, true).AnyTimes()
+		state.EXPECT().PulledContainerMapByArn(taskARN).Return(nil, true).AnyTimes()
+	}
+)
+
+// Builds and returns a new Task object with fault injection configurations
+func standardFaultInjectionTask() *apitask.Task {
+	task := apitask.Task{
+		Arn:                 taskARN,
+		Associations:        []apitask.Association{association},
+		Family:              family,
+		Version:             version,
+		DesiredStatusUnsafe: apitaskstatus.TaskRunning,
+		KnownStatusUnsafe:   apitaskstatus.TaskRunning,
+		NetworkMode:         apitask.HostNetworkMode,
+		ENIs: []*ni.NetworkInterface{
+			{
+				IPV4Addresses: []*ni.IPV4Address{
+					{
+						Address: eniIPv4Address,
+					},
+				},
+				MacAddress:               macAddress,
+				PrivateDNSName:           privateDNSName,
+				SubnetGatewayIPV4Address: subnetGatewayIpv4Address,
+			},
+		},
+		CPU:                      cpu,
+		Memory:                   memory,
+		PullStartedAtUnsafe:      now,
+		PullStoppedAtUnsafe:      now,
+		ExecutionStoppedAtUnsafe: now,
+		LaunchType:               "EC2",
+		NetworkNamespace:         defaultNamespace,
+		EnableFaultInjection:     true,
+	}
+	task.SetCredentialsID(taskCredentialsID)
+	return &task
+}
+
+// This function starts the server and listens on a specified port
+func startServer(t *testing.T) (*http.Server, int) {
+	router := mux.NewRouter()
+
+	// Mocks
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	state := mock_dockerstate.NewMockTaskEngineState(ctrl)
+	agentStateExpectations(state)
+	statsEngine := mock_stats.NewMockEngine(ctrl)
+	ecsClient := mock_ecs.NewMockECSClient(ctrl)
+
+	agentState := agentV4.NewTMDSAgentState(state, statsEngine, ecsClient, clusterName, availabilityzone, vpcID, containerInstanceArn)
+	metricsFactory := metrics.NewNopEntryFactory()
+	execWrapper := execwrapper.NewExec()
+
+	registerFaultHandlers(router, agentState, metricsFactory, execWrapper)
+
+	server := &http.Server{
+		Addr:    ":0", // Lets the system allocate an available port
+		Handler: router,
+	}
+
+	listener, err := net.Listen("tcp", server.Addr)
+	require.NoError(t, err)
+
+	port := listener.Addr().(*net.TCPAddr).Port
+	t.Logf("Server started on port: %d", port)
+
+	go func() {
+		if err := server.Serve(listener); err != nil && err != http.ErrServerClosed {
+			t.Logf("ListenAndServe(): %s\n", err)
+		}
+	}()
+	return server, port
+}
+
+// This function shuts down the server after the test
+func shutdownServer(t *testing.T, server *http.Server) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := server.Shutdown(ctx); err != nil {
+		fmt.Println("Server shutdown fail")
+		t.Logf("Server Shutdown Failed:%+v", err)
```

Comment:
Do we need `fmt.Println` or is `t.Logf` enough?

Reply:
Oops, that was a left over from manual testing. Yep t.Logf is enough

Reply:
removed